src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -163,22 +164,7 @@ public static void MatrixTimesSource(bool transpose, AlignedArray matrix, Aligne
         public static void Add(float value, Span<float> destination)
         {
             Contracts.AssertNonEmpty(destination);
-
-            if (destination.Length < MinInputSize || !Sse.IsSupported)
-            {
-                for (int i = 0; i < destination.Length; i++)
-                {
-                    destination[i] += value;
-                }
-            }
-            else if (Avx.IsSupported)
-            {
-                AvxIntrinsics.AddScalarU(value, destination);
-            }
-            else
-            {
-                SseIntrinsics.AddScalarU(value, destination);
-            }
+            TensorPrimitives.Add(destination, value, destination);
```

Comment:
Where are these methods consumed? Should the call sites be updated to use TensorPrimitives directly so that eventually CpuMathUtils goes away completely?

Reply:
There are some cases where we could do that, but not all, so I don't think CpuMathUtils will ever fully go away. Unless we want to add all the very specific stuff from ML.NET into TensorPrimitives.